addons/website_sale/models/website_snippet_filter.py

```diff
@@ -196,20 +195,24 @@ def _get_products_latest_sold(self, website, limit, domain, **kwargs):
             ('company_id', '=', website.company_id.id),
             ('state', '=', 'sale'),
         ], limit=8, order='date_order DESC')
+        delivery_carriers = self.env['delivery.carrier'].sudo().search([])
         if sale_orders:
             if self.env.context.get('hide_variants'):
-                sold_products = Counter(
+                sold_products = [
                     sol.product_id.product_tmpl_id.product_variant_id
                     for sol in sale_orders.order_line
-                )
+                ]
             else:
-                sold_products = Counter(sol.product_id for sol in sale_orders.order_line)
+                sold_products = [sol.product_id for sol in sale_orders.order_line]
```

Comment:
Maybe the use of `mapped` would be better here useful here 
```suggestion
                sold_products = sale_orders.mapped('order_line.product_id.product_tmpl_id.product_variant_id')
            else:
                sold_products = sale_orders.mapped('order_line.product_id')
```